andela/mmdp_site
# MMDP-309 Implement feature to manage reports

**What does this PR do?**
- Implements both backend and frontend to manage the report section feature.

**Description of Task to be completed?**

- [x] Admin should be able to view a grid list of reports.
- [x] Reports grid list should include pagination.
- [x] Admin should be able to edit a report.
- [x] Admin should be able to archive a report.
- [x] Admin should be able to unarchive a report.
- [x] Admin should be able to permanently delete the report.
- [x] Add tests for all features
- [x] Fixes events test which used to timeout.

**How should this be manually tested?**

- Clone this branch `ft-manage-report-MMDP-309`
- Run `yarn dev` to start both backend and frontend
- Navigate to ` localhost:1234/resources/reports` in your browser to test the frontend implementation
- List of reports should be displayed with pagination
- Edit link should navigate to the edit form page
- Delete, Archive and Unarchive actions should be functional when clicked

**Any background context you want to provide?**

[Zeplin mockup](https://zpl.io/adr98kl)

**What are the relevant Jira issues?**

[MMDP-309](https://viisaus.atlassian.net/browse/MMDP-309)
[MMDP-311](https://viisaus.atlassian.net/browse/MMDP-311)
[MMDP-312](https://viisaus.atlassian.net/browse/MMDP-312)
[MMDP-313](https://viisaus.atlassian.net/browse/MMDP-313)
[MMDP-314](https://viisaus.atlassian.net/browse/MMDP-314)
[MMDP-321](https://viisaus.atlassian.net/browse/MMDP-321)

**Screenshots (if appropriate)**
<img width="1440" alt="screen shot 2019-02-13 at 4 21 26 am" src="https://user-images.githubusercontent.com/9039613/52684631-2f0bd980-2f47-11e9-9dd1-b55b98313fd5.png">
<img width="1440" alt="screen shot 2019-02-13 at 4 21 35 am" src="https://user-images.githubusercontent.com/9039613/52684641-36cb7e00-2f47-11e9-9ecc-7ad800d2329e.png">
<img width="1440" alt="screen shot 2019-02-13 at 4 21 41 am" src="https://user-images.githubusercontent.com/9039613/52684647-3af79b80-2f47-11e9-9ae5-e1f783af0c1e.png">
<img width="1440" alt="screen shot 2019-02-13 at 4 21 50 am" src="https://user-images.githubusercontent.com/9039613/52684656-434fd680-2f47-11e9-8aed-032f98c0642c.png">


**Questions:**
